src/server/game/Server/Packets/MiscPackets.cpp

```diff
@@ -451,3 +451,74 @@ WorldPacket const* WorldPackets::Misc::Dismount::Write()
 
     return &_worldPacket;
 }
+
+void WorldPackets::Misc::SaveCUFProfiles::Read()
+{
+    uint8 strLen;
+
+    _worldPacket >> Count;
+
+    if (Count > MAX_CUF_PROFILES)
+        return;
+
+    for (uint8 i = 0; i < Count; i++)
+    {
+        CUFProfiles[i] = new CUFProfile;
```

Comment:
leaks memory if packet content is manipulated to be too short resulting in ByteBufferException being thrown - std::unique_ptr would be a good thing to use here